src/Data/List/Linear.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Data.List.Linear
```

Comment:
Could you add explicit export lists? Also, maybe make an issue to copy/paste important haddoc from base itself.